src/routes/cache-hooks/cache-hooks.controller.spec.ts

```diff
@@ -407,9 +428,12 @@ describe('Post Hook Events (Unit)', () => {
       .post(`/hooks/events`)
       .set('Authorization', `Basic ${authToken}`)
       .send(data)
-      .expect(200);
+      .expect(202);
 
-    await expect(fakeCacheService.get(cacheDir)).resolves.toBeNull();
+    setTimeout(
+      () => expect(fakeCacheService.get(cacheDir)).resolves.toBeNull(),
+      webHookExecutionDelayMs,
```

Comment:
This would add the assertion to the event loop which I don't think it's guaranteed to work after `webHookExecutionDelayMs`.

Did you try to use with `jest.advanceTimersByTime`?

Reply:
> This would add the assertion to the event loop which I don't think it's guaranteed to work after `webHookExecutionDelayMs`.

Could you elaborate on why this is not guaranteed to work? 

> Did you try to use with `jest.advanceTimersByTime`?

I saw it in the documentation, but it only works in the context of jest mocks, right? `fakeCacheService` is not a jest mock in this class. I'm not following on how to use that here, could you elaborate on that?

Reply:
> Could you elaborate on why this is not guaranteed to work?

The assertion will be added to the event loop but that doesn't mean that it will be executed exactly at `webHookExecutionDelayMs`. It will be executed at the next event cycle. I believe there can be a use case where the deletion can happen after the assertion. 
 
> I saw it in the documentation, but it only works in the context of jest mocks, right? `fakeCacheService` is not a jest mock in this class. I'm not following on how to use that here, could you elaborate on that?

Not really. Jest replaces the timers with fake ones. It is not related with mocks. See https://jestjs.io/docs/timer-mocks

Reply:
Synced internally, thanks for your comments @fmrsabino ! 🙂